bond_app/authentication.py

```diff
@@ -48,21 +49,23 @@ def __init__(self, accepted_audience_prefixes, accepted_email_suffixes, max_toke
 
 
 class Authentication:
-    def __init__(self, config, cache_api):
+    def __init__(self, config, cache_api, sam_api):
         """
         :param config: An AuthenticationConfig instance.
         :param cache_api: A CacheApi instance.
         """
         self.config = config
         self.cache_api = cache_api
+        self.sam_api = sam_api
 
     def require_user_info(self, request_state, token_info_fn=_token_info):
```

Comment:
Should we rename this?  `authn_authz_user`, `authorize_user`, `verify_access_token`, `validate_user`, ... ?

Reply:
I was trying to think of a good name because I don't think `require_user_info` is quite right anymore, but didn't come up with anything so just left it as is. How about `authenticate_user`? We aren't really doing any authz imo, we're just translating the token into the Sam user id

Reply:
Fine with me.   I would say we are doing the bare minimum authz in that we are checking to see that the user is enabled in Sam, so maybe we fudge it and go with `auth_user` and the reader can choose-their-own-adventure as to whether they think of it as `authz` or `authn` in their own brains?  🤣